### DISCUSSION TITLE
Update existing cluster via terraform?

### DISCUSSION THREAD

**Discussion by Lennix** [kube-hetzner/terraform-hcloud-kube-hetzner#607] [Q&A]
I don't know how to name this question properly.

I've stumbled upon a fix for kured to add the proper labels after rebooting.

How would I apply the fixed terraform ressources to my existing cluster?

This also includes a new parameter for kured in the kube.tf. Can I apply that as well?

**Comment by Lennix** [comment]
Steps I've taken so far:

1. Update terraform ressources via `terraform init --upgrade`
2. Apply via `terraform apply` which did change some kube-hetzner stuff (like firewall or labels) but it did not update any k3s values / kubernetes deployments (like kured)

**Reply by aleksasiriski** [reply]
I think you need to change the kustomize file and then run `terraform apply`

**Reply by mysticaltech** [reply]
@aleksasiriski Even that won't do it IMHO.

@Lennix To fix kured this is not needed, please see my other reply to you. However, just for knowledge, if you ever want to force reapply all settings that are part of kustomizations. You need to "terraform destroy -target null_resource.kustomization" and "terraform apply" again.

**Reply by ianwesleyarmstrong** [reply]
Would it be possible to add a [trigger](https://registry.terraform.io/providers/hashicorp/null/latest/docs/resources/resource#triggers) on `null_resource.kustomization`? It feels like a confusing experience to modify input of a module and not have them the model change anything.

**Reply by mysticaltech** [reply]
@ianwesleyarmstrong You are right! I think it's completely possible. Please do not hesitate to give it a shot and submit a PR if it works! 🚀

**Reply by mysticaltech** [reply]
We actually started doing this for one of the variable `k3s_registries`, and it's true that the experience is super smooth.
